File: 백준/Silver/10828. 스택/스택.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const N = Number(input[0]);
let stack = [];
let result = '';

for(let i=1; i<=N; i++) {
    let [cmd, ...rest] = input[i].split(' ');
    const stackLength = stack.length;
    
    switch(cmd) {
        case 'push':
            const num = Number(rest[0]);
            stack.push(num);
            break;
        case 'pop':
            const popNum = stackLength ? stack.pop() : -1;
            result += `${popNum}\n`;
            break;
        case 'size':
            result += `${stackLength}\n`;
            break;
        case 'empty':
            result += `${stackLength ? 0 : 1}\n`;
            break;
        default:
            result += `${stackLength ? stack[stackLength-1] : -1}\n`;
            break;
    }
}

console.log(result);

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

N = int(input().strip())
stack = []

for _ in range(N):
    command = input().strip().split()
    
    if command[0] == 'push':
        stack.append(int(command[1]))
    elif command[0] == 'pop':
        print(stack.pop() if stack else -1)
    elif command[0] == 'size':
        print(len(stack))
    elif command[0] == 'empty':
        print(0 if stack else 1)
    else:
        print(stack[-1] if len(stack) else -1) 